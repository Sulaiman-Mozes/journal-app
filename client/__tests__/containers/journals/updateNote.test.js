import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import UpdateNotePage, { UpdateNote } from '../../../containers/journals/updateNote';

describe('UpdateNotePage Container', () => {
  const middlewares = [thunk];
  const mockStore = configureStore(middlewares);
  const initialState = {};
  const store = mockStore(initialState);

  const fakeEvent = {
    target: { name: '', value: '' },
    preventDefault: jest.fn(),
  };

  const props = {
    update: { updateSuccess: 'update success' },
    history: { push: jest.fn() },
    setUpdateSuccessToFalseFunc: jest.fn(),
    getSingleNoteFunc: jest.fn(),
    updateDataFunc: jest.fn(),
    updateNoteFunc: jest.fn(),
    addNoteFunc: jest.fn(),
    match: { params: { id: '64738ewuyifhsjknvue8739' } },
    data: { title: '', content: '' },
  };

  it('should render correctly', () => {
    const updateNoteContainer = shallow(
      <Provider store={store}>
        <UpdateNotePage {...props} />
      </Provider>,
    );
    expect(updateNoteContainer).toMatchSnapshot();
  });

  it('should have the following methods', () => {
    const wrapper = shallow(<UpdateNote {...props} />);
    expect(wrapper.instance().handleChange(fakeEvent));
    expect(wrapper.instance().handleSubmit(fakeEvent));
  });
});
