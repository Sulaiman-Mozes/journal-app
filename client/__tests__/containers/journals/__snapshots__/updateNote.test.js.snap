// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateNotePage Container should render correctly 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeState": Object {},
    }
  }
>
  <Connect(UpdateNote)
    addNoteFunc={[MockFunction]}
    data={
      Object {
        "content": "",
        "title": "",
      }
    }
    getSingleNoteFunc={[MockFunction]}
    history={
      Object {
        "push": [MockFunction],
      }
    }
    match={
      Object {
        "params": Object {
          "id": "64738ewuyifhsjknvue8739",
        },
      }
    }
    setUpdateSuccessToFalseFunc={[MockFunction]}
    update={
      Object {
        "updateSuccess": "update success",
      }
    }
    updateDataFunc={[MockFunction]}
    updateNoteFunc={[MockFunction]}
  />
</ContextProvider>
`;
